### ISSUE TITLE
/var dataset requires legacy mount on systemd based distros.

### ISSUE THREAD

**Issue by kgunders** [zfsonlinux/zfs#3768]
When separating out /var and NOT using legacy mounts, journald writes to /var before zfs has initiated/mounted it.  Hence, zfs will subsequently be unable to mount /var, as it now contains existing data.  Note that I've not tested this very thoroughly, and it has been some years since I used zfs (so maybe I'm out in left field here....) but that sure looks to be what is happening.

**Comment by ryao** [comment]
systemd will need something like 3f1cc17c9005d19faba91fe7236332ed502b1a3f to address this.

**Comment by mailinglists35** [comment]
is this still an issue, right? I've just tried to migrate /var to zfs on centos7 but it does not get mounted in initrd process. if this is no longer an issue, what is the correct approach to migrate just /var from ext4 to zfs?

**Comment by sylveon** [comment]
I do not have this issue, but then my initrd mounts the /var partition before systemd even gets a chance to start.

**Comment by mailinglists35** [comment]
do you mind sharing your setup? i tried installing zfs-dracut and rebuilding initrd but systemd kicks in the initrd before zfs-import-cache

**Comment by mailinglists35** [comment]
@rlaager thank you. is it planned for next 0.7.x release or will be later in 0.8.0?

**Comment by zh99998** [comment]
does it work now?
I still can't mount /var without legacy mount.

debian buster, zfs 0.8.2

**Comment by rlaager** [comment]
@zh99998 Yes, but you have to be using the zfs-mount-generator, which is not enabled by default.

See step 5.8 here, starting with `mkdir /etc/zfs/zfs-list.cache`:
https://github.com/zfsonlinux/zfs/wiki/Debian-Buster-Root-on-ZFS
